sdcm/nemesis.py

```diff
@@ -1661,6 +1662,52 @@ def disrupt_nodetool_enospc(self, sleep_time=30, all_nodes=False):
                     finally:
                         clean_enospc_on_node(target_node=node, sleep_time=sleep_time)
 
+    def disrupt_end_of_quota_nemesis(self, sleep_time=30, all_nodes=False):
+        """
+        Nemesis flow
+        ---------------------
+        1. Check quota is enabled in the SCT configuration file.
+        2. Verify quota is enabled on the node.
+        3. Check disk usage.
+        4. Define quota size same as disk usage + 1GB.
+        5. Wait for end of quota message to appear in the log - "Storage I/O error: 122: Disk quota exceeded"
+        6. Except I/O Error if happens.
+        7. Remove the quota by setting it to 0
+        8. Restart scylla server
+        9. Verify scylla is up
+        """
+        if not self.cluster.params.get("enable_quota"):
+            raise UnsupportedNemesis("Quota configuration is not enabled on the cluster thus skipping this Nemesis")
+
+        if all_nodes:
+            nodes = self.cluster.nodes
+            InfoEvent('EDQUOT test on {}'.format([n.name for n in nodes])).publish()
+        else:
+            nodes = [self.target_node]
+
+        for node in nodes:
+            if self._is_it_on_kubernetes():
+                LOGGER.info("Skipping nemesis for kubernetes")
+                break
+            node.verify_quota_is_enabled_on_node()
+            with ignore_disk_quota_exceeded_errors(node=node):
+                result = self.target_node.remoter.run('cat /proc/mounts')
+                if '/var/lib/scylla' not in result.stdout:
+                    LOGGER.error("Scylla doesn't use an individual storage, skip edquota test")
+                    continue
+
+                try:
+                    configure_quota_on_node(target_node=node)
+                    wait.wait_for_log_lines(node=node, start_line_patterns='Disk quota exceeded',
+                                            end_line_patterns='Will retry in 3200ms',
+                                            start_timeout=60, end_timeout=600)
```

Comment:
The time 10min and size 1gb, are highly dependent on the test case, in some it would take more then 10min to fill 1gb.

Anyhow it's not clear what the end message you are looking for, it's a bit vague, and not clear why it's needed.

Reply:
Anyway, the `wait_for_log_lines` didn't work at all. 
In the last run it jumped right away to the next method `clean_edquota_on_node` so something it wrong here.

Reply:
I believe it's a context manger, so you are using it the wrong way

Reply:
Yep, I saw that already. Thank you for the input.